moto/ecr/models.py

```diff
@@ -486,12 +510,31 @@ def describe_images(self, repository_name, registry_id=None, image_ids=None):
 
         return response
 
-    def put_image(self, repository_name, image_manifest, image_tag):
+    def put_image(
+        self, repository_name, image_manifest, image_tag, image_manifest_mediatype=None
+    ):
         if repository_name in self.repositories:
             repository = self.repositories[repository_name]
         else:
             raise Exception(f"{repository_name} is not a repository")
 
+        try:
+            parsed_image_manifest = json.loads(image_manifest)
+        except json.JSONDecodeError:
+            raise Exception(
+                "Invalid parameter at 'ImageManifest' failed to satisfy constraint: 'Invalid JSON syntax'"
+            )
+
+        if image_manifest_mediatype:
+            parsed_image_manifest["imageManifest"] = image_manifest_mediatype
+        else:
+            if "mediaType" not in parsed_image_manifest:
+                raise Exception(
+                    "image manifest mediatype not provided in manifest or parameter"
+                )
+            else:
+                image_manifest_mediatype = parsed_image_manifest["mediaType"]
```

Comment:
This is set of changes approximates the behavior of AWS. The manifest media type is not a required parameter of the boto3 call, because AWS makes a strong assumption that the media type will be included within the image manifest, and will subsequently derive it from there. If it's not in the image manifest, the manifest is invalid.

I didn't go down the rabbit hole of edge cases surrounding this and simply implemented a coercion of given manifest media type to override. What does AWS do when you give a manifest list media type that differs from the one within the manifest? I do not know. It's a bunch of work to figure out for a case that represents an anti-pattern as far as I can see.  It's curious that such a scenario is even exposed as a possibility and I have a strong feeling this is an oddity of ECR.